indico/cli/i18n.py

```diff
@@ -113,79 +248,261 @@ def _wrapper(**kwargs):
 
+@cli.group('compile', short_help='Catalog compilation command for indico and indico plugins.')
+def compile_catalog():
+    """Catalog compilation command for indico and indico plugins."""
 
-cmd_list = ['init_catalog', 'extract_messages', 'update_catalog', 'compile_catalog',
-            'init_catalog_js', 'extract_messages_js', 'update_catalog_js',
-            'init_catalog_react', 'update_catalog_react']
 
+@cli.group('extract', short_help='Message extraction command for indico and indico plugins.')
+def extract_messages():
+    """Message extraction command for indico and indico plugins."""
 
-for cmd_name in cmd_list:
-    cli.add_command(_make_command(cmd_name))
 
+@cli.group('init', short_help='New catalog initialization command for indico and indico plugins.')
+def init_catalog():
+    """Catalog initialization command for indico and indico plugin.
 
-@cli.command()
-def extract_messages_react():
-    output = subprocess.check_output(['npx', 'react-jsx-i18n', 'extract', 'indico/web/client/', 'indico/modules/'],
-                                     env=dict(os.environ, FORCE_COLOR='1'))
-    with open(MESSAGES_REACT_POT, 'wb') as f:
-        f.write(output)
+    The sub commands take the new locale to be initialized as argument.
+    """
 
 
-@cli.command()
-def compile_catalog_react():
-    for locale in os.listdir(TRANSLATIONS_DIR):
-        po_file = os.path.join(TRANSLATIONS_DIR, locale, 'LC_MESSAGES', 'messages-react.po')
-        json_file = os.path.join(TRANSLATIONS_DIR, locale, 'LC_MESSAGES', 'messages-react.json')
-        if not os.path.exists(po_file):
-            continue
-        output = subprocess.check_output(['npx', 'react-jsx-i18n', 'compile', po_file])
-        json.loads(output)  # just to be sure the JSON is valid
-        with open(json_file, 'wb') as f:
-            f.write(output)
+@cli.group('update', short_help='Catalog merging command for indico and indico plugins.')
+def update_catalog():
+    """Catalog merging command for indico and indico plugins."""
+
+
+def _indico_command(ctx, babel_cmd, python, javascript, react, locale, no_check):  # pylint: disable=too-many-arguments
+    extra = {}
+    if locale:
+        extra['locale'] = locale
+    try:
+        if python:
+            _run_command(babel_cmd, extra=extra)
+        if javascript:
+            _run_command(f'{babel_cmd}_js', extra=extra)
+        if react:
+            if babel_cmd == 'compile_catalog':
+                compile_catalog_react()
+            elif babel_cmd == 'extract_messages':
+                extract_messages_react()
+            else:
+                _run_command(f'{babel_cmd}_react', extra=extra)
+    except Exception as err:
+        click.echo(click.style(f'Error running {babel_cmd} for indico - {err}', fg='red', bold=True), err=True)
+    if not no_check:
+        ctx.invoke(check_format_strings)
+
+
+def _plugin_command(ctx, babel_cmd, python, javascript, react, locale, no_check, plugin_dir):  # pylint: disable=R0913
```

Comment:
We do not use pylint in the core

```suggestion
def _plugin_command(ctx, babel_cmd, python, javascript, react, locale, no_check, plugin_dir):
```